translation.rst

```diff
@@ -827,6 +827,62 @@ checks translation resources for several locales:
     add the missing translation to the log file. For details,
     see :ref:`reference-framework-translator-logging`.
 
+Switch Locale Programmatically
+------------------------------
+
+.. versionadded:: 6.1
+
+    The ``LocaleSwitcher`` was introduced in Symfony 6.1.
+
+Sometimes you need to change the locale of the application dynamically
+just to run some code. Imagine a console command that renders Twig templates
+of emails in different languages. You need to change the locale only to
+render those templates.
+
+The ``LocaleSwitcher`` object/service allows you to change at once the locale
+of:
+
+1. All the services that are tagged with ``kernel.locale_aware``.
+2. ``\Locale::setDefault()``.
+3. If a request is available, the ``_locale`` request attribute.
+
+The ``LocaleSwitcher`` service is autowireable and can be injected into other
+services::
```

Comment:
I would remove this bit and add a small paragraph below the example, saying that the service will be autowired when using the default config and noting the service ID to use if that's not the case.